x-pack/legacy/plugins/apm/server/lib/settings/cm/save_configuration.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Setup } from '../../helpers/setup_request';
+import { PromiseReturnType } from '../../../../typings/common';
+import { CentralConfigurationIntake } from './configuration';
+
+export type CMSaveConfigurationAPIResponse = PromiseReturnType<
+  typeof saveConfiguration
+>;
+export async function saveConfiguration({
+  configuration,
+  setup
+}: {
+  configuration: CentralConfigurationIntake;
+  setup: Setup;
+}) {
+  const { client, config } = setup;
```

Comment:
should this endpoint not do its own validation of whether a configuration exists for a given serviceName/environment? Seems like a stale UI could fairly easily allow the user to create one.

Reply:
I can handle this in a different PR https://github.com/elastic/kibana/issues/40114